core/amber/src/main/scala/edu/uci/ics/amber/engine/common/ambermessage/WorkerMessage.scala

```diff
@@ -14,32 +14,23 @@ import edu.uci.ics.amber.engine.common.ambermessage.neo.{
 }
 import edu.uci.ics.amber.engine.common.ambertag.neo.VirtualIdentity
 import edu.uci.ics.amber.engine.common.ambertag.neo.VirtualIdentity.ActorVirtualIdentity
-import edu.uci.ics.amber.engine.common.promise.PromiseContext
 
 object WorkerMessage {
 
   final case class AckedWorkerInitialization(recoveryInformation: Seq[(Long, Long)] = Nil)
 
   final case class UpdateInputLinking(fromLayer: VirtualIdentity, inputNum: Int)
-      extends ControlPayload {
-    // TODO: use new promise API to send it
-    override val context: PromiseContext = null
-  }
+      extends ControlPayload
```

Comment:
Shoudnt these be extensions of ControlCommand? ControlPayload extensions are ControlInvocations and ReturnPayload.

Reply:
Since we haven't written their handling logic using the new API, for compatibility, I marked them as ControlPayload for now.

Reply:
Ok. Please add a TODO to change this later?